### PR TITLE
Drop pypy3.7, add wheels for pypy3.10

### PR DESCRIPTION
**Pull request by Starbuck5**

### CODE REVIEW

**Review by ankith26 — COMMENTED** [review]
I think we can drop PyPy 3.8 too (based on the download stats we discussed on discord, we don't really have users on it)

Also, the current distribution on mac is really suboptimal. Since PyPy builds are much slower in comparison to CPython builds, builds with PyPy have to be really minimal (it is fine to bundle 3-4 Cpython builds in one job, but don't bundle 2 PyPy builds in one)

**Comment by Starbuck5** [comment]
> Also, the current distribution on mac is really suboptimal. Since PyPy builds are much slower in comparison to CPython builds, builds with PyPy have to be really minimal (it is fine to bundle 3-4 Cpython builds in one job, but don't bundle 2 PyPy builds in one)

Ok, I'll rearrange it. Maybe back closer to how it was before.
I was trying to balance by number of jobs in general, which is why I split the arm64 build.

**Comment by Starbuck5** [comment]
I did the one thing you said not to, I put two pypy builds in the same run.

But I think it works out okay leveling everything off.

**Review by ankith26 — APPROVED** [review]
LGTM, thanks :partying_face:

File: .github/workflows/build-macos.yml [hunk 1]
```diff
@@ -85,35 +85,35 @@ jobs:
         # of these builds take roughly the same time
         include:
           - { 
-            name: "x86_64 (CPython 3.10 and above)",
+            name: "x86_64 (CPython 3.9 - 3.12)",
             macarch: x86_64,
             # pattern matches any 2 digit number
```
Comment (Matiiss) [inline .github/workflows/build-macos.yml:90]: This comment doesn't seem to be accurate anymore

Comment (Starbuck5) [reply .github/workflows/build-macos.yml]: Fixed.

**Review by Matiiss — APPROVED** [review]
LGTM

**Review by dr0id — COMMENTED** [review]
I'm not expert but why not add just such a snippet for each version added or remove one snippet instead of to have to adapt that many lines every time?

**Comment by Starbuck5** [comment]
> I'm not expert but why not add just such a snippet for each version added or remove one snippet instead of to have to adapt that many lines every time?

@dr0id Are you referring the macOS changes?

It would've been a simpler diff but I also needed to rebalance the runs so all take similar times. There are 5 runs because that's the maximum concurrent we can do with Github Actions.

**Pull request merged**